rclpy/src/rclpy/utils.cpp

```diff
@@ -89,4 +97,103 @@ convert_to_py(void * message, py::object pyclass)
   return py::reinterpret_steal<py::object>(convert(message));
 }
 
+const char *
+get_rmw_implementation_identifier()
+{
+  return rmw_get_implementation_identifier();
+}
+
+void
+assert_liveliness(py::object pyentity)
+{
+  if (PyCapsule_IsValid(pyentity.ptr(), "rclpy_publisher_t")) {
+    auto publisher = static_cast<rclpy_publisher_t *>(
+      rclpy_handle_get_pointer_from_capsule(
+        pyentity.ptr(), "rclpy_publisher_t"));
+    if (RCL_RET_OK != rcl_publisher_assert_liveliness(&publisher->publisher)) {
+      throw RCLError("Failed to assert liveliness on the Publisher");
+    }
+  } else {
+    throw UnsupportedObjectTypeError("Passed capsule is not a valid Publisher.");
+  }
+}
+
+py::list
+remove_ros_args(py::object pycli_args)
+{
+  rcl_ret_t ret;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_arguments_t parsed_args = rcl_get_zero_initialized_arguments();
+
+  RCPPUTILS_SCOPE_EXIT(
+    {
+      ret = rcl_arguments_fini(&parsed_args);
+      if (RCL_RET_OK != ret) {
+        throw RCLError("Failed to fini parsed args");
+      }
+    });
+
+  std::vector<const char *> arg_values;
+  const char ** const_arg_values = NULL;
+  py::list pyargs;
+  if (!pycli_args.is_none()) {
+    pyargs = pycli_args;
+    arg_values.resize(pyargs.size());
+    for (size_t i = 0; i < pyargs.size(); ++i) {
+      // CPython owns const char * memory - no need to free it
+      arg_values[i] = PyUnicode_AsUTF8(pyargs[i].ptr());
+      if (!arg_values[i]) {
+        throw py::error_already_set();
+      }
+    }
+    const_arg_values = &(arg_values[0]);
+  }
+
+  if (arg_values.size() > static_cast<size_t>(std::numeric_limits<int>::max())) {
+    throw py::value_error("too many cli arguments given to node");
+  }
+  int num_args = static_cast<int>(arg_values.size());
+  ret = rcl_parse_arguments(num_args, const_arg_values, allocator, &parsed_args);
+
+  if (RCL_RET_INVALID_ROS_ARGS == ret) {
+    throw RCLInvalidROSArgsError("Failed to parse ROS arguments");
+  }
+  if (RCL_RET_OK != ret) {
+    throw RCLError("failed to parse arguments");
+  }
+
+  throw_if_unparsed_ros_args(pyargs, parsed_args);
+
+  int nonros_argc = 0;
+  const char ** nonros_argv = NULL;
+
+  ret = rcl_remove_ros_arguments(
+    const_arg_values,
+    &parsed_args,
+    allocator,
+    &nonros_argc,
+    &nonros_argv);
+  if (RCL_RET_OK != ret) {
+    throw RCLError("Failed to init");
```

Comment:
Improve the error message
```suggestion
    throw RCLError("Failed rcl_remove_ros_arguments");
```